test/core-tests.el

```diff
@@ -586,6 +586,10 @@
       (setq header-line-format "aaaaaaa")
       (should (= (company--row) 0)))))
 
+;; Avoid compilation warnings on Emacs 25.
+(declare-function display-line-numbers-mode "ext:display-line-numbers")
```

Comment:
On the second thought, could it be safer for Emacs > 25 having this line wrapped into the Emacs-25 only version check? (I mean I'm not fully sure such function declaration is safe for the versions that have `display-line-numbers` file).

Reply:
They are safe enough: the declaration doesn't make the function defined, it just removes the byte-compilation warning.

Reply:
Such declarations are also used when you don't want to `require` a dependency outright, but load it on demand later when some function is called.

If a function belonging to said dependency is used that is not autoloaded, the file will get byte-compilation errors, which are handled with `declare-function`.

Reply:
BTW, I think people have been using the `ext:` prefix to mean third-party code (like externally distributed packages), and `display-line-numbers` is in the core.

Not that the file name matter much, though.